File: typescript/src/services/readjust/readjust-service.ts
import { AdjustValidation } from "../../interfaces/adjust-validation";
import { Employee } from "../../models/employee";

export class ReadjustService {
  private validations: AdjustValidation[];

  constructor(validations: AdjustValidation[]) {
    this.validations = validations;
  }

  public readjustSalary(employee: Employee, increase: number) {
    this.validations.forEach((validation) =>
      validation.validate(employee, increase)
    );

    const readjustedSalary = employee.personalData.salary + increase;
    employee.readjustSalary(readjustedSalary);
  }
}

File: java/src/main/java/br/com/rh/services/readjust/ReadjustService.java
package br.com.rh.services.readjust;

import br.com.rh.interfaces.AdjustValidation;
import br.com.rh.models.Employee;

import java.math.BigDecimal;
import java.util.List;

public class ReadjustService {
    private List<AdjustValidation> validations;

    public ReadjustService(List<AdjustValidation> validations) {
        this.validations = validations;
    }

    public void readjustSalary(Employee employee, BigDecimal increase) {
        this.validations.forEach(validation -> validation.validate(employee, increase));

        var readjustedSalary = employee.getPersonalData().getSalary().add(increase);
        employee.readjustSalary(readjustedSalary);
    }
}
